critiquebrainz/ws/review/views.py

```diff
@@ -367,12 +369,17 @@ def review_list_handler():
 
     # TODO(roman): Ideally caching logic should live inside the model. Otherwise it
     # becomes hard to track all this stuff.
-    cache_key = cache.gen_key('list', entity_id, user_id, sort, limit, offset, language)
-    cached_result = cache.get(cache_key, REVIEW_CACHE_NAMESPACE)
+    cache_key = None
+    cached_result = None
+
+    # If entity_id is absent, it is not possible to track cache keys. Hence, skip caching altogether in that case.
```

Comment:
I recall talking to you about this when you made the change, but I can't remember now - why can't we do it if there is no entity_id?
Is it because we create the `cache.gen_key('ws_cache', entity_id)` key to contain a list of all caches related to this entity_id?
What's the behaviour of this view if there is no entity_id? Is it a list of many different reviews? Why can't we have a key `ws_cache_None` (or even special-cased to `ws_cache_listkeys` or similar)  that contains the keys for this  non-id list view?

Reply:
`entity_id` is just another clause in the `WHERE`.
https://github.com/metabrainz/critiquebrainz/blob/master/critiquebrainz/db/review.py#L351-L353

The actual SQL is a few lines below. So the output depends on the rest of filters passed. Yeah, special casing is possible. I do not remember why we decided not to do that.

Reply:
I guess before adding redis sets this would have been a bit difficult. With redis sets, its easier. We can probably do this now.